autogen/agentchat/groupchat.py

```diff
@@ -79,26 +96,42 @@ def select_speaker(self, last_speaker: Agent, selector: ConversableAgent):
                         f"No agent can execute the function {self.messages[-1]['name']}. "
                         "Please check the function_map of the agents."
                     )
+
+        # remove the last speaker from the list to avoid selecting the same speaker if allow_repeat_speaker is False
+        agents = agents if self.allow_repeat_speaker else [agent for agent in agents if agent != last_speaker]
+
+        if self.speaker_selection_method == "manual":
+            print("Please select the next speaker from the following list:")
+            _n_agents = len(agents)
+            for i in range(_n_agents):
+                print(f"{i+1}: {agents[i].name}")
+            while True:
+                try:
+                    i = int(input("Enter the number of the next speaker: "))
+                    if i > 0 and i <= _n_agents:
+                        return agents[i - 1]
+                    else:
+                        raise ValueError
+                except ValueError:
+                    print(f"Invalid input. Please enter a number between 1 and {_n_agents}.")
+        elif self.speaker_selection_method == "round_robin":
+            return self.next_agent(last_speaker, agents)
+        elif self.speaker_selection_method == "random":
+            return random.choice(agents)
         else:
```

Comment:
Do we want to consider asserting that 
`speaker_selection_method in ['auto','manual','random','round_robin']`

so that 'MANUAL' (note uppercase) won't fall into this else statement (unintentionally by the developer).